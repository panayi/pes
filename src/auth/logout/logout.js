import * as R from 'ramda';
import { firebaseConnect } from 'react-redux-firebase';
import { defaultProps, withProps } from 'recompose';
import omitProps from '../../lib/helpers/omitProps';
import noop from '../../lib/helpers/noop';

export const logout = R.compose(
  defaultProps({
    onLogout: noop,
  }),
  withProps(({ firebase, onLogout }) => ({
    onClick: () => {
      firebase.logout();
      onLogout();

      // FIXME: react-redux-firebase clears data on logout
      // This is a workaround to get the data back
      window.location.reload();
    },
  })),
  omitProps(['onLogout']),
);

export default R.compose(
  firebaseConnect(),
  logout,
);
