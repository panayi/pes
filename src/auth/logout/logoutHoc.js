import R from 'ramda';
import { firebaseConnect } from 'react-redux-firebase';
import { defaultProps, withProps } from 'recompose';
import noop from '../../lib/helpers/noop';

export default R.compose(
  firebaseConnect(),
  defaultProps({
    onLogout: noop,
  }),
  withProps(({ firebase, onLogout }) => ({
    onClick: () => {
      firebase.logout();
      onLogout();

      // FIXME: react-redux-firebase clears data on logout
      // This is a workaround to get the data back
      window.location.reload();
    },
  })),
);
