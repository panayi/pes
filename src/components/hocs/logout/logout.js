import * as R from 'ramda';
import { firebaseConnect } from 'react-redux-firebase';
import { defaultProps, withProps } from 'recompose';
import omitProps from 'utils/omitProps';
import noop from 'utils/noop';

export const logout = R.compose(
  defaultProps({
    onLogout: noop,
  }),
  withProps(({ firebase, onLogout }) => ({
    onClick: () => {
      firebase.logout();
      onLogout();

      // FIXME: react-redux-firebase clears data on logout
      // This is a workaround to get the data back
      window.location.reload();
    },
  })),
  omitProps(['onLogout']),
);

export default R.compose(
  firebaseConnect(),
  logout,
);
