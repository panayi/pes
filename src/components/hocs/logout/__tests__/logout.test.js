import React from 'react';
import { logout } from '../logout';

describe('logout', () => {
  it('should render correctly', () => {
    const firebase = {
      logout: noop,
    };
    const Button = logout(() => (
      <button>
        Log out
      </button>
    ));
    const wrapper = mount(
      <Button firebase={firebase} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should call firebase.logout on click', () => {
    const mockLogout = jest.fn();
    const firebase = {
      logout: mockLogout,
    };
    const LogoutButton = 'button';
    const Hoc = logout(LogoutButton);
    const mockOnLogout = jest.fn();
    const wrapper = mount(
      <Hoc
        firebase={firebase}
        onLogout={mockOnLogout}
      />,
    );

    expect(mockLogout.mock.calls.length).toBe(0);
    expect(mockOnLogout.mock.calls.length).toBe(0);

    wrapper.find(LogoutButton).simulate('click');

    expect(mockLogout.mock.calls.length).toBe(1);
    expect(mockOnLogout.mock.calls.length).toBe(1);
  });
});
