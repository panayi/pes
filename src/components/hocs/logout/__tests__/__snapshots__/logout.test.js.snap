// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logout should render correctly 1`] = `
<defaultProps(withProps(mapProps(Component)))
  firebase={
    Object {
      "logout": [Function],
    }
  }
  onLogout={[Function]}
>
  <withProps(mapProps(Component))
    firebase={
      Object {
        "logout": [Function],
      }
    }
    onLogout={[Function]}
  >
    <mapProps(Component)
      firebase={
        Object {
          "logout": [Function],
        }
      }
      onClick={[Function]}
      onLogout={[Function]}
    >
      <Component
        firebase={
          Object {
            "logout": [Function],
          }
        }
        onClick={[Function]}
      >
        <button>
          Log out
        </button>
      </Component>
    </mapProps(Component)>
  </withProps(mapProps(Component))>
</defaultProps(withProps(mapProps(Component)))>
`;
